codalab/worker/rest_client.py

```diff
@@ -160,7 +163,21 @@ def _upload_with_chunked_encoding(
             conn.send(b'0\r\n\r\n')
 
             # Read the response.
-            response = conn.getresponse()
+            logging.debug("About to read the response... url: %s", url)
+
+            # Sometimes, it may take a while for the server to process
+            # the data and send the response. In this case, we want to
+            # periodically keep sending empty bytes so that the
+            # connection doesn't drop before the response is available.
+            got_response = False
+            while not got_response:
```

Comment:
Any chance this results in a infinite loop? Feeling at some point we should fail the request after a set number of tries

Reply:
We're actually already in an infinite loop with the code from `master` -- `conn.getresponse()` just hangs forever, which is causing the error. Since this PR is just a strict improvement to what was there earlier, I'd rather just do the bare minimum to fix the problem for now.

Reply:
Sounds good